server/shared/publishing/helpers.js

```diff
@@ -171,6 +172,8 @@ function saveFile(parent, key, data) {
 }
 
 function saveSpine(spine) {
+  if (spine.version) delete spine.version;
+  spine.version = hasha(JSON.stringify(spine), { algorithm: 'sha1' });
```

Comment:
`JSON.stringify` does not do object keys sorting thus two identical spines with different properties insertion order would result in different hashes 🛑 
Instead of relaying on fact that spines are always generated in same way i.e. having same properties insertion order we could use safer alternative: [hash-obj](https://npm.im/hash-obj) 🚑